base/linalg/lapack.jl

```diff
@@ -3847,28 +3869,36 @@ for (syconv, hesv, hetrf, hetri, hetrs, elty, relty) in
 # *     .. Array Arguments ..
 #       INTEGER            IPIV( * )
 #       COMPLEX*16         A( LDA, * ), B( LDB, * )
-        function hetrs!(uplo::Char, A::StridedMatrix{$elty},
-                       ipiv::Vector{BlasInt}, B::StridedVecOrMat{$elty})
-            chkstride1(A,B)
-            n = chksquare(A)
-            if n != size(B,1)
-                throw(DimensionMismatch("B has first dimension $(size(B,1)), but needs $n"))
+            function $trsname(uplo::Char, A::StridedMatrix{$elty},
+                              ipiv::Vector{BlasInt}, B::StridedVecOrMat{$elty})
+                chkstride1(A,B)
+                n = chksquare(A)
+                if n != size(B,1)
+                    throw(DimensionMismatch("B has first dimension $(size(B,1)), but needs $n"))
+                end
+                info  = Array(BlasInt, 1)
+                ccall(($(blasfunc(hetrs)), liblapack), Void,
+                      (Ptr{UInt8}, Ptr{BlasInt}, Ptr{BlasInt}, Ptr{$elty}, Ptr{BlasInt},
+                       Ptr{BlasInt}, Ptr{$elty}, Ptr{BlasInt}, Ptr{BlasInt}),
+                      &uplo, &n, &size(B,2), A, &max(1,stride(A,2)), ipiv, B, &max(1,stride(B,2)), info)
+                @lapackerror
+                B
             end
-            info  = Array(BlasInt, 1)
-            ccall(($(blasfunc(hetrs)), liblapack), Void,
-                  (Ptr{UInt8}, Ptr{BlasInt}, Ptr{BlasInt}, Ptr{$elty}, Ptr{BlasInt},
-                   Ptr{BlasInt}, Ptr{$elty}, Ptr{BlasInt}, Ptr{BlasInt}),
-                  &uplo, &n, &size(B,2), A, &max(1,stride(A,2)), ipiv, B, &max(1,stride(B,2)), info)
-            @lapackerror
-            B
         end
     end
 end
 
-for (sysv, sytrf, sytri, sytrs, elty, relty) in
-    ((:zsysv_,:zsytrf_,:zsytri_,:zsytrs_,:Complex128, :Float64),
-     (:csysv_,:csytrf_,:csytri_,:csytrs_,:Complex64, :Float32))
-    @eval begin
+for suffix in (symbol(""), :_rook)
+    for (sysv, sytrf, sytri, sytrs, elty, relty) in
```

Comment:
I think it would be simpler to have a loop for each function given the convoluted code just below.